src/Symfony/Component/Messenger/Adapter/PhpAmqp/AmqpReceiver.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Adapter\PhpAmqp;
+
+use Symfony\Component\Messenger\Adapter\PhpAmqp\Exception\RejectMessageExceptionInterface;
+use Symfony\Component\Messenger\Transport\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\DecoderInterface;
+
+/**
+ * Symfony Messenger receiver to get messages from AMQP brokers using PHP's AMQP extension.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class AmqpReceiver implements ReceiverInterface
+{
+    public const STOP = 'stop';
+
+    private $messageDecoder;
+    private $connection;
+
+    public function __construct(DecoderInterface $messageDecoder, Connection $connection)
+    {
+        $this->messageDecoder = $messageDecoder;
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(): iterable
+    {
+        $shouldStop = false;
+
+        while (!$shouldStop) {
+            if (null === $message = $this->connection->waitAndGet()) {
+                continue;
+            }
+
+            try {
+                $shouldStop = self::STOP === (yield $this->messageDecoder->decode(array(
+                    'body' => $message->getBody(),
+                    'headers' => $message->getHeaders(),
+                )));
+
+                $this->connection->ack($message);
+            } catch (RejectMessageExceptionInterface $e) {
+                $this->connection->reject($message);
+
+                throw $e;
+            } catch (\Throwable $e) {
+                $this->connection->nack($message);
```

Comment:
I would requeue the message, because if you don't, the message will be lost

Reply:
Updated 👍